Added devfile registry docs (#73)
devfile/docs: docs/modules/user-guide/partials/proc_devfile-registry.adoc

> [id="proc_devfile-registry_{context}"]

Reviewer: * Don't add the module type to the ID (regardless of what the current templates show).
* This is not a procedure module. The file needs to be renamed, and the Antora page fixed.

```suggestion
[id="devfile-registry_{context}"]
```

= Devfile registry

[role="_abstract"]

> A devfile registry is a service that stores and provides devfile stacks to Kubernetes developer tools like odo, Eclipse Che and the OpenShift Developer Console. Therefore, you can access devfiles through the devfile registry.  Devfile registries are based on the Oracle Cloud Infrastructure (OCI) Specification, and devfile stacks are stored as OCI-artifacts in the registry.

Reviewer 2: ```suggestion
A devfile registry is a service that stores and provides devfile stacks to Kubernetes developer tools like odo, Eclipse Che, and the OpenShift Developer Console. Therefore, you can access devfiles through the devfile registry.  Devfile registries are based on the Oracle Cloud Infrastructure (OCI) Specification, and devfile stacks are stored as OCI-artifacts in the registry.
```

> Each devfile stack corresponds to a specific runtime or framework, such as NodeJS, Quarkus, or Wildfly. A devfile stack also contains resources like a `devfile.yaml` file, logo, and outerloop resources. These devfile stacks give developers a template to get started developing cloud-native applications.

Reviewer: @jc-berger,

* Always check project/product naming.
* "Outerloop" needs an explanation.

```suggestion
Each devfile stack corresponds to a specific runtime or framework, such as Node.js, Quarkus, or WildFly. A devfile stack also contains resources like a `devfile.yaml` file, logo, and outerloop resources. These devfile stacks provide developers with templates to get started developing cloud-native applications.
```

> Each devfile stack corresponds to a specific runtime or framework, such as,  NodeJS, Quarkus, and Wildfly. A devfile stack also contains resources, such as, a devfile.yaml, logo, and outerloop resources. These devfile stacks give developers a template so they can get started developing cloud-native applications.

Reviewer 2: Is "outerloop" Red Hat jargon or is it a devfile term that is well-understood users?

Author: Good point. @johnmcollier, I'm thinking users will be unfamiliar with `logo` and `outerloop resources`; therefore, maybe we should remove them. What do you think?

> Each devfile stack corresponds to a specific runtime or framework, such as,  NodeJS, Quarkus, and Wildfly. A devfile stack also contains resources, such as, a devfile.yaml, logo, and outerloop resources. These devfile stacks give developers a template so they can get started developing cloud-native applications.

Reviewer 2: ```suggestion
Each devfile stack corresponds to a specific runtime or framework, such as NodeJS, Quarkus, or Wildfly. A devfile stack also contains resources like a `devfile.yaml` file, logo, and outerloop resources. These devfile stacks give developers a template to get started developing cloud-native applications.
```

== Devfile registry components

> A devfile registry has two components: a devfile index server and an OCI registry server.

Reviewer: ```suggestion
A devfile registry has two components:

 * Devfile index server
 * OCI registry server
```

.Devfile index container

The devfile index container image does the following actions:

> * Uses the index.json file to obtain metadata about the stacks and samples and hosts this metadata in the registry for tools to consume.

Reviewer: ```suggestion
* Uses the `index.json` file to obtain metadata about the stacks and samples and hosts this metadata in the registry for tools to consume.
```

* Bootstraps the OCI registry with the devfile stacks.
* Provides an API to interact with the OCI registry and to retrieve devfile stacks.

.OCI registry

The devfile stacks in a devfile registry are stored in an OCI-compatible artifact registry. The artifact registry is based on the reference implementation of an OCI registry.

When a devfile registry is deployed, the stacks belonging to the devfile registry are pushed to the OCI registry as multi-layer artifacts.

== Types of devfile registries

> As a developer, you can use multiple types of devfile registries to access devfile stacks and tools.

Reviewer: ```suggestion
There are multiple types of devfile registries to access devfile stacks and tools.
```

.Public community registry

> The public community registry provides stacks for various community tools to consume, such as NodeJS, Quarkus, and Open Liberty.

Reviewer: ```suggestion
The public community registry provides stacks for various community tools to consume, such as Node.js, Quarkus, and Open Liberty.
```

> The public community registry provides stacks for various community tools to consume. Examples include: NodeJS, Quarkus, and Open Liberty.

Reviewer 2: ```suggestion
The public community registry provides stacks for various community tools to consume, such as NodeJS, Quarkus, and Open Liberty.
```
IBM SG: "Use a colon after an _independent clause_ to introduce an inline list."

> Locate the source repository for the stacks in the public community registry at the devfile/registry repository.

Reviewer: ```suggestion
Locate the source for the stacks in the public community registry in the link:https://github.com/devfile/registry[devfile/registry repository].
```

.On-cluster devfile registry

> Private, on-cluster devfile registries can be configured on your own Kubernetes cluster, deployed using either the devfile registry operator or the devfile registry helm chart.

Reviewer: ```suggestion
Devfile registries can be operated on private Kubernetes clusters, deployed using either the devfile registry Operator or the devfile registry Helm Chart.
```

> The on-cluster devfile registries can be configured with custom devfile stacks suited to your needs for your organization.

Reviewer: ```suggestion
The on-cluster devfile registries can be configured with custom devfile stacks suited to the needs of the owner organization.
```

== Additional resources

> After understanding the basics of a devfile registry, start working with a devfile registry by going to xref:authoring-stacks.adoc[].

Reviewer: ```suggestion
* xref:authoring-stacks.adoc[].
```

[id="proc_devfile-registry_{context}"]
= Devfile registry

[role="_abstract"]

> A devfile registry is a service that stores and provides devfile stacks to Kubernetes developer tools like odo, Eclipse Che and the OpenShift Developer Console. Therefore, users can access devfiles through the devfile registry.  Devfile registries are based on the Oracle Cloud Infrastructure (OCI) Specification, and devfile stacks are stored as OCI-artifacts in the registry.

Reviewer 2: ```suggestion
A devfile registry is a service that stores and provides devfile stacks to Kubernetes developer tools like odo, Eclipse Che, and the OpenShift Developer Console. Therefore, users can access devfiles through the devfile registry. Devfile registries are based on the Oracle Cloud Infrastructure (OCI) Specification and devfile stacks are stored as OCI-artifacts in the registry.
```
IBM SG: "Use a serial comma (comma before and or or for three or more items)."

> Each devfile stack corresponds to a specific runtime or framework, such as,  NodeJS, Quarkus, and Wildfly. A devfile stack also contains resources, such as, a devfile.yaml, logo, and outerloop resources. These devfile stacks give users a template so they can get started developing with devfiles. For more information on the structure of a devfile registry, see https://github.com/johnmcollier/registry-docs/blob/main/registry-structure.adoc[registry-structure].

Author: I wanted to end on a statement targeted towards the users. So I said users can use devfile stacks as a template to get started with devfiles. That's correct, right?

Author: Good call! Made the change, thanks!

> Each devfile stack corresponds to a specific runtime or framework, such as,  NodeJS, Quarkus, and Wildfly. A devfile stack also contains resources, such as, a devfile.yaml, logo, and outerloop resources. These devfile stacks give users a template so they can get started developing with devfiles. For more information on the structure of a devfile registry, see https://github.com/johnmcollier/registry-docs/blob/main/registry-structure.adoc[registry-structure].

Author: As for the links, once we get all the devfile registry docs included in our docs repo, they can be xrefs (because they'll be linking to our own docs). For now, they'll have to be outside webpages that we're linking to.

Author: I was hoping you'd say that because that's my thought exactly! In the new commit, I'll remove the links, thanks!

> Each devfile stack corresponds to a specific runtime or framework, such as,  NodeJS, Quarkus, and Wildfly. A devfile stack also contains resources, such as, a devfile.yaml, logo, and outerloop resources. These devfile stacks give users a template so they can get started developing with devfiles. For more information on the structure of a devfile registry, see https://github.com/johnmcollier/registry-docs/blob/main/registry-structure.adoc[registry-structure].

Reviewer 2: ```suggestion
Each devfile stack corresponds to a specific runtime or framework such as NodeJS, Quarkus, and Wildfly. A devfile stack also contains resources such as a `devfile.yaml`, logo, and outerloop resources. These devfile stacks give users a template to get started developing with devfiles. For more information on the structure of a devfile registry, see https://github.com/johnmcollier/registry-docs/blob/main/registry-structure.adoc[registry-structure].
```

- You might need to get approval to publish a link to a GitHub repo. 
- Use `link:` with external links.

Author: I removed the link, so it's no longer a concern, thanks. But it's good to know!

== Devfile registry components

> A devfile registry is composed of two components, a devfile index server and an OCI registry server.

Reviewer 2: ```suggestion
A devfile registry has two components: a devfile index server and an OCI registry server.
```

Reviewer 2: IBM SG: "Use a colon after an independent clause to introduce an inline list."

.Devfile index server

> The server, and the resulting container image, does the following:

Author: The original version was, 
`The server (and resulting container image) that does the following`

Could we say, `the server and *its* container do the following` 

I don't want to change meaning...also, if it's put in a parentheses, maybe it's not pertinent enough to include?

Author: Good revision. Made the change, thanks!

> The server, and the resulting container image, does the following:

Reviewer 2: IBM SG: "**following** adjective, noun
To introduce a list, procedure, or subsequent items, always use "following" as an adjective before a noun. For example, do not write "Complete the following" or "The functions include the following". Instead, write
"Complete the following steps" or "The adapter has the following functions"."

Author: Thanks, Rolfe, I've made the change, let me know what you think, thanks!

> * Uses the index.json file to obtain metadata about the stacks and samples and hosts this metadata in the registry for tools to consume

Author: The original first bullet point was: 
`Hosts metadata about the stacks and samples in the registry for tools to consume (via the index.json file).`

I found that hard to digest. Is this revision good and correct?

> * Bootstraps the OCI registry with the devfile stacks
> * Provides an API to interact with the OCI registry and to retrieve devfile stacks

Reviewer 2: ```suggestion
* Bootstraps the OCI registry with the devfile stacks.
* Provides an API to interact with the OCI registry and to retrieve devfile stacks.
```

Reviewer 2: IBM SG: **Punctuation in lists**
Write lists so that all or none of the items start with complete sentences. If list items comprise only complete sentences, include a period after each sentence. Except in definition lists, do not include any end punctuation if list items comprise only sentence fragments.
If a list contains list items that start with sentence fragments, and one or more of the fragments are followed by complete sentences, include periods after all sentence fragments and sentences.

> For more information on the index server, see https://github.com/johnmcollier/registry-docs/blob/main/registry-REST-API.adoc[registry-REST-API].

Reviewer 2: Use `link:`

Author: Thanks, Rolfe, I've removed all the links. There's no only one xref under `additional resources`. I'll know to do this next time though when I have external links to include!

.OCI registry

> The devfile stacks in a devfile registry are stored in an OCI-compatible artifact registry, currently based on the https://github.com/distribution/distribution[reference implementation of an OCI registry].

Reviewer 2: ```suggestion
The devfile stacks in a devfile registry are stored in an OCI-compatible artifact registry. The artifact registry is based on this link:https://github.com/distribution/distribution[reference implementation of an OCI registry].
```
- Broke this up to make it easier to read. To clarify _what_ is based on the reference implementation.
- You might have problems getting permission to link to GitHub repos. 
- Use `link:`

Author: Thanks, made the change!

> When a devfile registry is deployed, the stacks belonging to the devfile registry are pushed to the OCI registry as multi-layer artifacts. For more information on the structure of the devfile stacks within the OCI registry, see https://github.com/johnmcollier/registry-docs/blob/main/registry-structure.adoc[registry-structure].

Author: Any opinion on how many links we have in this doc? I personally think we have too many. I think when there's this many links, it becomes _very easy_ for users to leave our page and fall down a rabbit hole. I wanted a second opinion. If anyone agrees and has an opinion on which links to eliminate, I'm all ears, thanks!

== Types of devfile registries

> Users can use multiple types of devfile registries to access devfile stacks and tools.

Reviewer 2: ```suggestion
As an <role>, you can use multiple types of devfile registries to access devfile stacks and tools.
```
- IBM SG: Write clearly and directly. Wherever possible, use the second person ("you"). 
- If it helps, you can also identify the role of the user. For example "As an administrator, you..."

Author: Thanks for the suggestion! Made the change, thanks!

> .Public community registry

Reviewer 3: These are just showing up as plain text in the rendered doc rather than a subheader. Should they be subheaders?

Author: They look like subheaders to me. The screenshot I've included is what I see on my local server. Does this match with what you see?

<img width="1300" alt="Screen Shot 2021-06-07 at 3 46 13 PM" src="https://user-images.githubusercontent.com/70717303/121078905-7f0daa80-c7a7-11eb-8eaa-b711ef888827.png">

Reviewer 3: Okay that looks fine to me, must be a quirk with how GitHub renders the pages. So long as they look good on the website, that's good 👍

The public community registry provides stacks for various community tools to consume. Examples include: https://registry.devfile.io/devfiles/nodejs[NodeJS], https://registry.devfile.io/devfiles/java-quarkus[Quarkus], and https://registry.devfile.io/devfiles/java-openliberty[Open Liberty].

Locate the source repository for the stacks in the public community registry at the https://github.com/devfile/registry[devfile/registry repository].

.On-cluster devfile registry

> Private, on-cluster devfile registries can be configured on a user's own Kubernetes cluster, deployed using either the https://github.com/devfile/registry-operator[devfile registry operator] or the https://github.com/devfile/registry-support/tree/master/deploy/chart/devfile-registry[devfile registry helm chart].

Reviewer 2: Review content for multiple instances of "the user" or "a user's" that you can replace with "you" and "your."
